projects/firmware_homework102/src/main.c

```diff
@@ -0,0 +1,60 @@
+#include <stdint.h>
+#include <stdlib.h>
+#include <time.h>
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+};
+
+struct Counters *counter_init() {
+  struct Counters *out = malloc(sizeof(struct Counters));
+  out->counter_a = 0;
+  out->counter_b = 0;
+  return out;
+}
+
+void delay(unsigned milliseconds) {
+  clock_t pause;
+  clock_t start;
+
+  pause = milliseconds * (CLOCKS_PER_SEC / 1000);
+  start = clock();
+  while ((clock() - start) < pause) {
+    continue;
+  }
+}
+
+void increment(SoftTimerId timer_id, void *context) {
+  struct Counters *temp = context;
+  temp->counter_a += 1;
+  LOG_DEBUG("Counter A: %d\n", temp->counter_a);
+  delay(0.5);
```

Comment:
This is delaying for 0ms (`(unsigned)0.5`), not 500ms, eh? In any case, it's fine for this homework, but in real non-unit test firmware using `delay` isn't advised because it blocks execution. Usually we have several tasks we'd like each board to do, and using `delay` delays all of them.